src/Rings/mpoly.jl

```diff
@@ -207,12 +201,96 @@ mutable struct BiPolyArray{S}
     r = new{elem_type(T)}(Ox)
     r.Sx = Sx
     r.S = S
+    return r
+  end
+end
+
+mutable struct IdealGens{S}
+  gens::BiPolyArray{S}
+  isGB::Bool
+  ord::Orderings.MonomialOrdering
+  keep_ordering::Bool
+
+  function IdealGens(O::Vector{T}; keep_ordering::Bool = true, isGB::Bool = false) where {T <: NCRingElem}
+    return IdealGens(parent(O[1]), O; keep_ordering = keep_ordering,
+                                        isGB = isGB)
+  end
+
+  function IdealGens(Ox::NCRing, O::Vector{T}; keep_ordering::Bool = true, isGB::Bool = false) where {T <: NCRingElem}
+    r = new{T}()
+    r.gens = BiPolyArray(Ox, O)
+    r.isGB = isGB
+    r.keep_ordering = keep_ordering
```

Comment:
But `ord` should be also set if the input is coming from an Oscar array.

Reply:
You mean by an additional argument?

Reply:
If possible exactly like the corresponding function with Singular input. That one works without an extra argument.

Reply:
There the ordering comes from the Singular side. But here I don’t know from where to get the ordering. 
Nevertheless, in the current state the function is wrong since there shouldn’t be any case where `isGB` is true but there is no ordering.

Reply:
Well, then the question is: Do we allow instantiations of `IdealGens` without `ord` being set? I would assume "no" since this is one of the main features of `IdealGens`.

Reply:
We allow the ordering to not being set, since the data structure is also for plain generating systems. If isGB is true then an ordering has to be set. If the input comes from an OSCAR array then the ordering has to be explicitely specified. There should not be any situation where we know that something is a Gröbner basis, but do not know the respective ordering.